pkg/blockdevice/v1alpha1/blockdevice.go

```diff
@@ -229,13 +241,39 @@ func filterNonInactive(orignialList *BlockDeviceList) *BlockDeviceList {
 	return filteredList
 }
 
-func filterClaimedDevices(orignialList *BlockDeviceList) *BlockDeviceList {
+func filterNonPartitions(originalList *BlockDeviceList) *BlockDeviceList {
 	filteredList := &BlockDeviceList{
 		BlockDeviceList: &ndm.BlockDeviceList{},
 		errs:            nil,
 	}
-	for _, device := range orignialList.Items {
-		if device.Status.ClaimState == ndm.BlockDeviceClaimed {
+	for _, device := range originalList.Items {
+		if strings.EqualFold(device.Spec.Partitioned, "No") {
+			filteredList.Items = append(filteredList.Items, device)
+		}
+	}
+	return filteredList
+}
+
+func filterSparseDevices(originalList *BlockDeviceList) *BlockDeviceList {
+	filteredList := &BlockDeviceList{
+		BlockDeviceList: &ndm.BlockDeviceList{},
+		errs:            nil,
+	}
+	for _, device := range originalList.Items {
+		if strings.EqualFold(device.Spec.Details.DeviceType, string(apis.TypeSparseCPV)) {
```

Comment:
EqualFold will ignore the case letters ?

Reply:
IMO we can use direct string comparison .

Reply:
Updated but need to change whenever there is change at ndm side